### PR TITLE
Switch from reflect.DeepEqual to Semantic.DeepEqual

### PR DESCRIPTION
**Pull request by Pejdzor**
golang reflect.DeepEqual() sometime returns false when comparing 2 semantically identical objects, so we switch from using this to Semantic.DeepEqual().

### CODE REVIEW

File: controllers/resources/helper.go [hunk 1]
```diff
@@ -23,9 +23,9 @@ import (
 	"fmt"
 	"math/big"
 	"os"
+	"reflect"
 	"regexp"
 
-	"reflect"
```
Comment (ljeda) [inline controllers/resources/helper.go:28]: why you remove this package but add it two lines above?

**Review by ljeda — CHANGES_REQUESTED** [review]
also, did you run any automation for these changes? no regression?

File: controllers/resources/helper.go (hunk 1 above)
Comment (Pejdzor) [reply controllers/resources/helper.go:28]: Auto-refactoring from IDE, it moves imports between lines, if they suits more in other position.
In this situation it grouped built-in package imports above other (external) ones.

**Review by ljeda — APPROVED** [review]

**Review by kflorianski-ibm — APPROVED** [review]

**Pull request merged**